components/tyk-dashboard/templates/_helpers.tpl

```diff
@@ -55,6 +55,32 @@ http
 gateway-svc-{{.Release.Name}}-tyk-gateway.{{ .Release.Namespace }}.svc
 {{- end -}}
 
+{{/*
+    It lists all services in the release namespace and find a service
+    for Tyk Gateway with its label.
+*/}}
+{{- define "tyk-dashboard.gatewaySvcName" -}}
+   {{- $services := (lookup "v1" "Service" .Release.Namespace "") -}}
+   {{- if $services -}}
+       {{- range $index, $svc := $services.items -}}
+           {{- range $key, $val := $svc.metadata.labels -}}
+               {{- if and (eq $key "app") (contains "gateway-svc-" $val) -}}
```

Comment:
Should we check for control service too?
If control service exist use that otherwise gateway svc?